src/main/java/hudson/plugins/sonar/MsBuildSQRunnerEnd.java

```diff
@@ -0,0 +1,130 @@
+ *
+ * Jenkins Plugin for SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * Jenkins Plugin for SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+/*
+ * Sonar is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * Sonar is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with Sonar; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package hudson.plugins.sonar;
+
+import hudson.plugins.sonar.action.UrlSonarAction;
+import hudson.plugins.sonar.utils.SonarUtils;
+
+import hudson.plugins.sonar.action.BuildSonarAction;
+import hudson.model.Action;
+import hudson.plugins.sonar.action.ProjectSonarAction;
+import org.kohsuke.stapler.DataBoundConstructor;
+import jenkins.model.Jenkins;
+import hudson.AbortException;
+import hudson.EnvVars;
+import hudson.plugins.sonar.utils.BuilderUtils;
+import hudson.util.ArgumentListBuilder;
+import hudson.Extension;
+import hudson.model.AbstractProject;
+import hudson.tasks.BuildStepDescriptor;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.model.TaskListener;
+import hudson.model.Run;
+
+import java.io.IOException;
+
+import hudson.tasks.Builder;
+
+public class MsBuildSQRunnerEnd extends AbstractMsBuildSQRunner {
+  @DataBoundConstructor
+  public MsBuildSQRunnerEnd() {
+    // will use MSBuild SQ Runner installation defined in Begin
+  }
+
+  @Override
+  public void perform(Run<?, ?> run, FilePath workspace, Launcher launcher, TaskListener listener) throws InterruptedException, IOException {
+    ArgumentListBuilder args = new ArgumentListBuilder();
+
+    EnvVars env = BuilderUtils.getEnvAndBuildVars(run, listener);
+    String runnerName = loadRunnerName(env);
+    MsBuildSQRunnerInstallation msBuildRunner = Jenkins.getInstance().getDescriptorByType(MsBuildSQRunnerBegin.DescriptorImpl.class)
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'getInstance' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2259)